docs/tutorials/javascript/amm/trade-with-auction-slot.md

```diff
@@ -0,0 +1,649 @@
+
+```javascript
+    try {
+
+        const standby_wallet = xrpl.Wallet.fromSeed(standbySeedField.value)
+
+        const takerPaysCurrency = standbyTakerPaysCurrencyField.value
+        const takerPaysIssuer = standbyTakerPaysIssuerField.value
+        const takerPaysAmount = standbyTakerPaysAmountField.value
+
+        const takerGetsCurrency = standbyTakerGetsCurrencyField.value
+        const takerGetsIssuer = standbyTakerGetsIssuerField.value
+        const takerGetsAmount = standbyTakerGetsAmountField.value
+
+        let takerPays = null
+        let takerGets = null
+
+        if ( takerPaysCurrency == 'XRP' ) {
+            takerPays = xrpl.xrpToDrops(takerPaysAmount)
+        } else {
+            takerPays = {
+                "currency": takerPaysCurrency,
+                "issuer": takerPaysIssuer,
+                "value": takerPaysAmount
+            }
+        }
+
+        if ( takerGetsCurrency == 'XRP' ) {
+            takerGets = xrpl.xrpToDrops(takerGetsAmount)
+        } else {
+            takerGets = {
+                "currency": takerGetsCurrency,
+                "issuer": takerGetsIssuer,
+                "value": takerGetsAmount
+            }
+        }
+
+        results += '\n\nSwapping tokens ...'
+        standbyResultField.value = results
+```
+
+Submit the `OfferCreate` transaction.
+
+```javascript
+        const offer_result = await client.submitAndWait({
+            "TransactionType": "OfferCreate",
+            "Account": standby_wallet.address,
+            "TakerPays": takerPays,
+            "TakerGets": takerGets
+        }, {autofill: true, wallet: standby_wallet})
+        
+        if (offer_result.result.meta.TransactionResult == "tesSUCCESS") {
+            results += `\n\nTransaction succeeded.`
+            checkAMM()
+        } else {
+            results += `\n\nError sending transaction: ${JSON.stringify(offer_result.result.meta.TransactionResult, null, 2)}`
+        }        
+    } catch (error) {
+        results += `\n\n${error.message}`
+    }
+```
+
+Report the results.
+
+```javascript
+    standbyResultField.value = results
+
+    client.disconnect()
+
+}
+```
+
+
+## Code Walkthrough (ripplex13b-amm-formulas.js) 
+
+You can open `ripplex13b-amm-formulas.js` from the [Quickstart Samples](https://github.com/XRPLF/xrpl-dev-portal/tree/master/_code-samples/quickstart/js/) to view the source code. This tutorial uses three of the available [AMM helper functions](https://github.com/XRPLF/rippled/blob/2d1854f354ff8bb2b5671fd51252c5acd837c433/src/ripple/app/misc/impl/AMMHelpers.cpp).
+
+
+### swapOut()
+
+The `swapOut()` function calculates how much of an asset you must deposit into an AMM to receive a specified amount of the paired asset. The input asset is what you're adding to the pool (paying), and the output asset is what you're receiving from the pool (buying).
+
+```javascript
+/* Implement the AMM SwapOut formula, as defined in XLS-30 section 2.4 AMM 
```

Comment:
This whole comment in particular would work better to my mind as descriptive text, rather than inline commentary.